evergreen/compile.sh

```diff
@@ -9,5 +9,14 @@ set -o errexit  # Exit the script with error if any of the commands fail
 
 echo "Compiling .NET driver"
 
-for var in TMP TEMP NUGET_PACKAGES NUGET_HTTP_CACHE_PATH APPDATA; do setx $var z:\\data\\tmp; export $var=z:\\data\\tmp; done
-powershell.exe .\\build.ps1 -target Build -Verbosity Diagnostic
+if [ "$OSTYPE" = "cygwin" ]; then
+  for var in TMP TEMP NUGET_PACKAGES NUGET_HTTP_CACHE_PATH APPDATA; do
+    setx $var z:\\data\\tmp
+    export $var=z:\\data\\tmp
+  done
+  powershell.exe .\\build.ps1 -target Build -Verbosity Diagnostic
+else
+  for var in TMP TEMP NUGET_PACKAGES NUGET_HTTP_CACHE_PATH APPDATA; do
+    export $var=/data/tmp;
```

Comment:
Am I understanding the logic correctly that we only compile on Windows under cygwin? For Linux, we just use pre-built nuget packages? If so, I would suggest we echo "Skipping" or something to that effect in the else clause to avoid future confusion. Right now our output logs on Linux are going to say "Compiling .NET driver" with no indication that it isn't actually compiled on that platform.

Reply:
Actually you caught a bug. I've updated it to actually compile both.